### PR TITLE
ATS-250 : ATS: T-Engines: Improve the convert message error handling (T-Base)

### PR DESCRIPTION
**Pull request by DenisGabriela**
- code refactoring:
 -- if successful -> return the object ( Optional.ofNullable(request) )
 -- if not successful -> throw an exception (instead of sending an error message on the queue)
   - when calling 'convert()' method, catch this exception and send a T-Reply with error message on the queue

### CODE REVIEW

File: alfresco-transformer-base/src/main/java/org/alfresco/transformer/QueueTransformService.java [hunk 1]
```diff
@@ -94,75 +97,65 @@ public void receive(final Message msg)
...
+            return;
+        }
 
-        TransformRequest transformRequest = convert(msg, correlationId, replyToDestinationQueue);
-        if (transformRequest == null)
+        if (!transformRequest.isPresent())
         {
-            logger.error("Exception during T-Request deserialization! T-Reply with error has been "
-                + "sent to T-Router!");
+            logger.error("T-Request is null deserialization!");
```
Comment (montgolfiere) [inline alfresco-transformer-base/src/main/java/org/alfresco/transformer/QueueTransformService.java]: Error messages should contain some detailed context (correlationId, requestId, sourceReference, ...) to allow log grep / tracing.

File: alfresco-transformer-base/src/main/java/org/alfresco/transformer/QueueTransformService.java [hunk 2]
```diff
@@ -94,75 +97,65 @@ public void receive(final Message msg)
...
-                logger.error("T-Request is null deserialization!");
-                replyWithInternalSvErr(destination,
-                    "JMS exception during T-Request deserialization: ", correlationId);
-            }
-            return request;
+            return Optional.ofNullable(request);
         }
         catch (MessageConversionException e)
         {
             String message = "Message conversion exception during T-Request deserialization: ";
```
Comment (montgolfiere) [inline alfresco-transformer-base/src/main/java/org/alfresco/transformer/QueueTransformService.java]: See comment above - add T-Request context to error messages ?

File: alfresco-transformer-base/src/main/java/org/alfresco/transformer/QueueTransformService.java [hunk 3]
```diff
@@ -94,75 +97,65 @@ public void receive(final Message msg)
...
-            logger.error(message + e.getMessage(), e);
-            replyWithBadRequest(destination, message + e.getMessage(), correlationId);
+            throw new TransformException(HttpStatus.BAD_REQUEST.value(), message + e.getMessage());
         }
         catch (JMSException e)
         {
             String message = "JMS exception during T-Request deserialization: ";
```
Comment (montgolfiere) [inline alfresco-transformer-base/src/main/java/org/alfresco/transformer/QueueTransformService.java]: See comment above (add T-Request context to error messages).

**Comment by DenisGabriela** [comment]
@montgolfiere , thanks for review. I've updated the PR.

**Comment by DenisGabriela** [comment]
@montgolfiere, do you want to take another look on this before I merge it?

**Pull request merged**